tests/models/test_cat_model.py

```diff
@@ -297,3 +297,58 @@ async def test_find_many(
     )
 
     assert found_cat_summaries == expected_cat_summaries
+
+
+@pytest.mark.parametrize(
+    "existing_cat_documents, cat_id, expected_response",
+    [
+        (
+            [
+                {
+                    "_id": ObjectId("000000000000000000000101"),
+                    "name": "Sammybridge Cat",
+                    "ctime": datetime(2020, 1, 1, 0, 0, tzinfo=UTC),
+                    "mtime": datetime(2020, 1, 1, 0, 0, tzinfo=UTC),
+                },
+                {
+                    "_id": ObjectId("000000000000000000000102"),
+                    "name": "Shirasu Sleep Industries Cat",
+                    "ctime": datetime(2020, 1, 1, 0, 0, tzinfo=UTC),
+                    "mtime": datetime(2020, 1, 1, 0, 0, tzinfo=UTC),
+                },
+            ],
+            dto.CatID("000000000000000000000101"),
+            True,
+        ),
+        (
+            [
+                {
+                    "_id": ObjectId("000000000000000000000101"),
+                    "name": "Sammybridge Cat",
+                    "ctime": datetime(2020, 1, 1, 0, 0, tzinfo=UTC),
+                    "mtime": datetime(2020, 1, 1, 0, 0, tzinfo=UTC),
+                },
+                {
+                    "_id": ObjectId("000000000000000000000102"),
+                    "name": "Shirasu Sleep Industries Cat",
+                    "ctime": datetime(2020, 1, 1, 0, 0, tzinfo=UTC),
+                    "mtime": datetime(2020, 1, 1, 0, 0, tzinfo=UTC),
+                },
+            ],
+            dto.CatID("000000000000000000000000"),
+            False,
+        ),
+    ],
+)
+@conftest.async_test
+async def test_delete_one(
+    existing_cat_documents: List[BSONDocument],
+    cat_id: dto.CatID,
+    expected_response: bool,
+) -> None:
+    collection = await get_collection(cat_model._COLLECTION_NAME)
+    await collection.insert_many(existing_cat_documents)
+
+    result = await cat_model.delete_one(cat_id)
+
+    assert result == expected_response
```

Comment:
[suggestion] We also need to check whether the document(s) are actually from the DB. 
We can do this by retrieving all documents from the collection and check if this empty.
WDYT?

Reply:
@ujwalaprabhakar Sorry if my comment was a bit confusing. In case we originally had one cat in the DB and we delete said cat, we should also check if the cat is actually removed from the DB within this test. 
This test only checks the return result which does not guarantee that the cat is actually removed (i.e a function can "fake" the return value or some logic inside the function is wrong and it always returns `True`)
So in this test you can retrieve all of the documents from the `cats` collection and check if that is an empty list.
Also a test to check what will happen if no cat can be found is missing.
Please let me know if this clarifies things or if you still have some questions.

Reply:
I tried somthing. Please have a look.

Reply:
```suggestion
    actual_documents = [document async for document in collection.find()]

    assert result == expected_response
    assert actual_documents == expected_documents
```